bundle/edu.gemini.p2checker/src/main/java/edu/gemini/p2checker/rules/gems/GemsGuideStarRule.java

```diff
@@ -109,20 +107,29 @@ public IP2Problems check(ObservationElements elements)  {
 
             // Extract the CWFS targets and count them.
             GuideGroup primaryGuideGroup = env.getPrimaryGuideGroup();
-            final List<SiderealTarget> cwfsTargets = new ArrayList<>();
+            final Map<SiderealTarget,CanopusWfs> cwfsTargets = new HashMap<>();
             int cwfs = 0;
             for (final CanopusWfs canwfs : CanopusWfs.values()) {
-                Option<GuideProbeTargets> gpt = primaryGuideGroup.get(canwfs);
+                final Option<GuideProbeTargets> gpt = primaryGuideGroup.get(canwfs);
                 if (!gpt.isEmpty() && !gpt.getValue().getPrimary().isEmpty()) {
-                    gpt.getValue().getPrimary().foreach(t -> ImOption.fromScalaOpt(t.getSiderealTarget()).foreach(cwfsTargets::add));
+                    gpt.getValue().getPrimary().foreach(t -> ImOption.fromScalaOpt(t.getSiderealTarget()).foreach(st -> cwfsTargets.put(st, canwfs)));
                     cwfs++;
                 }
             }
+            final ImList<SiderealTarget> cwfsTargetsImList = DefaultImList.create(cwfsTargets.keySet());
+
+            // Check for overlapping guide windows.
+            CanopusWfs.Group.findOverlappingGuideWindows(ctx, cwfsTargetsImList).foreach(p -> {
+                final SiderealTarget t1 = p._1();
+                final SiderealTarget t2 = p._2();
+                final CanopusWfs c1 = cwfsTargets.get(t1);
+                final CanopusWfs c2 = cwfsTargets.get(t2);
+                problems.addError(PREFIX + "GuideWindowOverlapError", String.format(WindowOverlap, c1.getKey(), c2.getKey()), targetNode);
+            });
 
             // Check for magnitude violations.
-            final ImList<SiderealTarget> cwfsTargetsImList = DefaultImList.create(cwfsTargets);
-            final ImList<Double> cwfsMagnitudes = CanopusWfs.Group.instance.extractRMagnitudes(cwfsTargetsImList);
-            if (!cwfsTargets.isEmpty() && !cwfsMagnitudes.isEmpty() && !CanopusWfs.Group.instance.checkAsterismMagnitude(DefaultImList.create(cwfsTargets)))
+            final ImList<Double> cwfsMagnitudes = CanopusWfs.Group.extractRMagnitudes(cwfsTargetsImList);
+            if (!cwfsTargets.isEmpty() && !cwfsMagnitudes.isEmpty() && !CanopusWfs.Group.checkAsterismMagnitude(cwfsTargetsImList))
```

Comment:
Is it possible to have non-empty `cwfsMagnitudes` without `cwfsTargets`?

Reply:
This only came up when I just made some dummy manual targets and didn't specify any magnitudes for them. In that case, `CanopusWfs.Group.checkAsterismMagnitude` returns `false`. Without the check for non-empty `cwfsMagnitudes`, the error would be generated, and it didn't seem right to me to issue an error about magnitudes being too far from one another when there were no magnitudes specified at all.

Reply:
My question is whether we need both `!cwfsTargets.isEmpty() && !cwfsMagnitudes.isEmpty()`.  The magnitudes can only be non-empty if there are targets, right?

Reply:
Oh... ha! Yes. That's a good point. Okay, I'll push another commit to this PR and reopen it.